reflect.go

```diff
@@ -465,23 +464,39 @@ func recordedObjectString(obj types.Object) objectString {
 	return pkg.Path() + "." + obj.Name()
 }
 
+func reflectedObjectString(obj types.Object, parent *types.Struct) string {
+	if obj == nil {
+		return ""
+	}
+	pkg := obj.Pkg()
+	if pkg == nil {
+		return ""
+	}
+
+	if v, ok := obj.(*types.Var); ok && parent != nil {
+		return hashWithStruct(parent, v)
+	}
+
+	lpkg := sharedCache.ListedPackages[obj.Pkg().Path()]
+	return hashWithPackage(lpkg, obj.Name())
+
+}
```

Comment:
why can't we just use recordedObjectString here? if the two funcs are fundamentally different, please document that in the godoc here